library/src/main/java/com/getstream/sdk/chat/rest/controller/RetrofitClient.java

```diff
@@ -16,14 +19,32 @@
 
 public class RetrofitClient {
 
-    public static Retrofit getAuthorizedClient(CachedTokenProvider tokenProvider, ApiClientOptions options) {
+    private static HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor();
 
-        HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor();
-        loggingInterceptor.level(HttpLoggingInterceptor.Level.BODY);
+    private static Request prepareRequest(Interceptor.Chain chain, boolean isAuthorizedClient) {
+        String authType = isAuthorizedClient ? "jwt" : "anonymous";
 
-        TokenAuthInterceptor authInterceptor = new TokenAuthInterceptor(tokenProvider);
+        return chain.request()
+                .newBuilder()
+                .addHeader("Content-Type", "application/json")
+                .addHeader("stream-auth-type", authType)
+                .addHeader("Accept-Encoding", "application/gzip")
+                .build();
+    }
 
-        OkHttpClient client = new OkHttpClient.Builder()
+    public static Retrofit getClient(ApiClientOptions options, @Nullable CachedTokenProvider tokenProvider) {
```

Comment:
using null for selecting anon auth is not great; code gets harder to read and easier to break. I can think of a couple better options here:

- use overloading and code extraction (no breaking change + clear APIs to get normal or anon client)
- add anonymous bool argument to signature (breaking change + clear API + less code changes + needs some args validation)